src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -474,9 +474,13 @@ public boolean openPairingWindowWithPINCallback(
         deviceControllerPtr, devicePtr, duration, iteration, discriminator, setupPinCode, callback);
   }
 
-  /* Shutdown all cluster attribute subscriptions for a given device */
-  public void shutdownSubscriptions(long devicePtr) {
-    shutdownSubscriptions(deviceControllerPtr, devicePtr);
+  public long getFabricId() {
+    return getFabricId(deviceControllerPtr);
+  }
+
+  /* Shutdown subscription based upon peerNodeId, fabricIndex, and subscriptionId */
+  public void shutdownSubscription(long peerNodeId, int fabricIndex, long subscriptionId) {
```

Comment:
Should the fabric index calculation be done internally to the controller? I'm guessing it would not be valid to try to shut down a subscription from a different fabricIndex from the currently established session? Also, looking at the usage, shouldn't this be `fabricId` rather than `fabricIndex`? Those are conceptually different things.

Reply:
should be fabricIndex...would update across.

Reply:
updated, thanks